### DISCUSSION TITLE
"ERROR: KVM: error opening /dev/kvm: No such file or directory"

### DISCUSSION THREAD

**Discussion by emendelson** [dosemu2/dosemu2#1753] [General]
**Describe the bug**
Upgraded Ubuntu Mate to the latest version; removed and then reinstalled dosemu2. It starts from /usr/bin/dosemu, but with the error message about /dev/kdm shown in the heading of this report.

**To Reproduce**
Install dosemu2 from the repository, run /usr/bin/dosemu from the command line.

**Attach the log**
Attached:
[boot.log](https://github.com/dosemu2/dosemu2/files/9151808/boot.log)

**A regression?**
This error didn't occur until upgrade Ubuntu Mate to 2022.04 and reinstalled.

**Comment by stsp** [comment]
So you don't have /dev/kvm.
Its that simple. :)
What is the problem itself, besides
an msg?

**Comment by andrewbird** [comment]
I'm not familiar with 22.04 yet, but in the past, flavours of Ubuntu have needed the package `qemu-kvm` installed and a logout/login to enable `/dev/kvm`. Perhaps it's still the case?

**Comment by stsp** [comment]
I think it was not "No such file or
directory" but rather the "Access denied".
So I think its not related.

**Comment by jharrison03** [comment]
Make a udev rule to set group permissions on /dev/kvm and add your user to the KVM group --

/etc/udev/rules.d/65-kvm.rules
`KERNEL=="kvm", GROUP="kvm", MODE="0660"`

**Comment by stsp** [comment]
But "No such file or directory" is
not a permission problem.

**Comment by andrewbird** [comment]
No such file or directory on `/dev/kvm` means either the kvm module wasn't loaded or it didn't initialise successfully. The latter meaning either the cpu doesn't support virtualisation, or it's disabled in the BIOS. Remember that if you newly enable virtualisation in the BIOS, then a power off/on is required for it to become active (a restart is not enough)

**Comment by stsp** [comment]
Moved to discussions.

**Comment by nelgin** [comment]
For those of us running on a VPS without embedded virtualization, how do I get rid of this warning/error? Dosemu is being run from my BBS and therefore, I believe, no tty is used so I cannot redirect to /dev/null. 

Is /dev/kvm actually a requirement for dosemu2 to work?

**Reply by stsp** [reply]
You can use `dosemu -q` to suppress errors.

**Comment by nelgin** [comment]
Huh, I thought I tried that. Thank you.
